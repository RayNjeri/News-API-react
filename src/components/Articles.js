import React from 'react';
import Axios from 'axios';
import Articlesrender from './Articlesrender';

export default class Articles extends React.Component {
  constructor() {
    super();
    this.state = {
      articles: [],
    };
  }

  renderArticles(sourceId, sortBy) {
    let url = `https://newsapi.org/v1/articles?source=${sourceId}&sortBy=${sortBy}&apiKey=3e94e65c70b047a3971837eb342fff8b`;
    Axios.get(url)
      .then((result) => {
        this.setState({
          articles: result.data.articles
        });
      });

  }
  componentDidMount() {
    this.renderArticles(this.props.params.sourceId, this.props.params.sortBy);
  };
  componentWillReceiveProps(nextProps) {
    this.renderArticles(nextProps.params.sourceId, nextProps.params.sortBy);
  }
  render() {
    return (
      <Articlesrender articles={this.state.articles} />
    );
  }
};

