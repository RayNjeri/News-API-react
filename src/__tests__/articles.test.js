import React from 'react';
import { mount, shallow } from 'enzyme';
import shallowToJSON from 'enzyme-to-json';
import Articles from '../components/Articles';

describe('Articles Component', () => {
  it('Snapshot of Articles Component', () => {
    const articles = shallow(<Articles />);
    const tree = shallowToJSON(articles);
    expect(articles.containsMatchingElement(<h3>Articles</h3>)).toBe(true);
    expect(tree).toMatchSnapshot();
  });
});
