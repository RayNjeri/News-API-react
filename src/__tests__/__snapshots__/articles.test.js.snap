// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles Component Snapshot of Articles Component 1`] = `
<div
  className="articles col-lg-8"
>
  <h3>
    Articles
  </h3>
  <div
    className="row"
  />
</div>
`;
